src/convertToObject.js

```diff
@@ -5,8 +5,23 @@
  *
  * @return {object}
  */
+
 function convertToObject(sourceString) {
-  // write your code here
+  const parts = sourceString.split(';');
```

Comment:
The split operation on the `sourceString` is correct, but you should chain additional methods to handle the transformation in a more streamlined and functional manner.